### ISSUE TITLE
Only the @i values of Journal items are added to the tab completions list

### ISSUE THREAD

**Issue by therden** [dagraham/etm-dgraham#11]
Even after exiting and re-starting etm, the `@i` index value of non-Journal items are not added to the tab completion list.

**Comment by dagraham** [comment]
I doubt that enhancing support for index entry tab completion is the best
option but will think about it.

Dan

**Comment by therden** [comment]
Thank you Dan. 

If, as I hope, you _do_ decide to go that way, the versions of `get_completions()` and `update_completions()` which follow each capture and add the `@i` values of items of _all_ types to the completions list.

```
    def get_completions(self):
        """
        Get completions from db items
        """
        completions = set([])
        for item in self.db:
            found = {x: v for x, v in item.items() if x in self.completion_keys}

            for x, v in found.items():
                if isinstance(v, list):
                    if x == "k":
                        continue
                    for p in v:
                        completions.add(f"@{x} {p}")
                else:
                    if x == "i":
                        i, t, s, d = (
                            item["i"],
                            item["itemtype"],
                            item["summary"],
                            item.doc_id,
                        )
                        completions.add(f"@k {i} {t} {s}: {d}")
                    else:
                        completions.add(f"@{x} {v}")
        self.completions = list(completions)
        self.completions.sort()

    def update_completions(self, item):
        """
        update self.completions, if necessary, for the @keys in item
        """
        completions = set([])

        found = {x: v for x, v in item.item_hsh.items() if x in self.completion_keys}
        for x, v in found.items():
            # if x == 'k':
            if isinstance(v, list):
                for p in v:
                    completions.add(f"@{x} {p}")
            else:
                if x == "i":
                    i, t, s, d = (
                        v,
                        item.item_hsh.get("itemtype", None),
                        item.item_hsh.get("summary", None),
                        item.doc_id,
                    )
                    completions.add(f"@k {i} {t} {s}: {d}")
                else:
                    completions.add(f"@{x} {v}")
        new = [x for x in list(completions) if x not in self.completions]
        if new:
            self.completions.extend(new)
            self.completions.sort()

```

**Comment by dagraham** [comment]
This was implemented in 4.7.4.
